core/controllers/reader.py

```diff
@@ -111,30 +111,65 @@ def get(self, exploration_id):
 class ExplorationPage(base.BaseHandler):
     """Page describing a single exploration."""
 
+    URL_PATH_ARGS_SCHEMAS = {
+        'exploration_id': {
+            'schema': {
+                'type': 'basestring'
+            }
+        }
+    }
+    HANDLER_ARGS_SCHEMAS = {
+        'GET': {
+            'v': {
+                'schema': {
+                    'type': 'int'
+                },
+                'default_value': None
+            },
+            'parent': {
+                'schema': {
+                    'type': 'basestring'
+                },
+                'default_value': None
+            },
+            'iframed': {
+                'schema': {
+                    'type': 'basestring'
+                },
```

Comment:
Why is it basestring? It seems more like boolean to me.

Reply:
Same reason as apply_draft, the arguments coming from request are of type string/unicode, I can add choices  ```['true', 'false']``` but looking for your opinion.

Reply:
Can we change the frontend code to pass booleans directly? If not can we use something to modify the validator to accept only 'true'/'false' and then normalize them to booleans?

Reply:
According to [this discussion](https://github.com/oppia/oppia/pull/13370/files/d8018706b1b8d4f01c2de7dcc56e714191b68108#diff-3a5d5cbd63bf7a719c023d1c78bdfdbbbfd416e22716d96d874fd613326fa7f4) with @seanlip schema validation system (SVS) should normalize the "true"/"false" into corresponding boolean value, but as we depend entirely on schema utils for normalization of arguments we cannot modify schema_utils to do normalization of this data's.
Since python3 migration will make standard for an edge case condition as mentioned by me [here](https://github.com/oppia/oppia/pull/13370#discussion_r669459140).
Thus I am working on a PR for normalizing this type of args by schema validation system & then update my other PR's accordingly.

Reply:
Done in this PR, PTAL